analytics.go

```diff
@@ -453,12 +454,35 @@ func (c *client) send(msgs []message, retryAttempt int) {
 // Upload serialized batch message.
 func (c *client) upload(b []byte, targetNode string) error {
 	url := c.Endpoint + "/v1/batch"
-	req, err := http.NewRequest("POST", url, bytes.NewReader(b))
+	gzipPayload := func(data []byte) (io.Reader, error) {
```

Comment:
We don't want to break backwards compatibility with older versions of rudder-server. Therefore, the SDK should first make sure that the receiving end is able to accept gzip-compressed payloads. The same strategy must be followed for any backwards incompatible features we introduce in the future, i.e. the SDK shouldn't use a feature that rudder-server doesn't support.

The information on the features supported by rudder-server will be available in its `/version` endpoint.

Please see the relevant pull request in rudder-server [here](https://github.com/rudderlabs/rudder-server/pull/2751)

Reply:
We decided that we will not check for the version from the SDK. We will mention in the doc if a developer wants to upgrade to this version, it is their responsibility to use the rudder server version 1.4.x

Reply:
Then you should at least bump a major version with your next SDK release.

Can you please share the main reasons that led to such a decision?

Reply:
Reasons are below:
1. Parity with Segment.
2. Easy to implement.
3. The facility we provide shouldn’t be too strict.

Reply:
This implementation should be enough.